### DISCUSSION TITLE
Migration guide: switch `master` to `main` branch

### DISCUSSION THREAD

**Discussion by edgarriba** [kornia/kornia#2719] [General]
After some years of active development to `master` branch, we have migrated to `main` to fulfill Github standards.

:link:  https://www.theserverside.com/feature/Why-GitHub-renamed-its-master-branch-to-main

As a developer, what to do ?

- If have an open PR against the old branch, please tag any of the project maintainers to help you with the rebase.

**Comment by johnnv1** [comment]
As part of this migration, the entire commit history of the repository was rewritten to make it lighter. Using the [BFG Repo-Cleaner](https://rtyley.github.io/bfg-repo-cleaner/) we have deleted from the history some artifacts (files >= ~0.5Mb). [The complete list of these files can be checked on the commit message](https://github.com/kornia/kornia/commit/4808b331811aeb78381d7f6283a05a7447a8d37a).

Consequently, when all branches based on the old `master` branch are deleted, the kornia repository will move from ~88Mb to ~14Mb. We are planning to delete these branches (and the `master` itself) around the date of [the release v0.7.2 (end of march).](https://github.com/kornia/kornia/milestone/3)

As part of this effort, in oct of 2023, we removed the unused docs build branch (see https://github.com/kornia/kornia/pull/2624), which already moved the repository from ~122Mb to ~88Mb.

So, if you want to have a lightweight fork of Kornia, I recommend you delete all branches aside from the `main` (be careful not to delete active branches that you have a PR on).
